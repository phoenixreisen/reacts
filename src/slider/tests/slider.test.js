/**
 * @jest-environment jsdom
 */
import {render, fireEvent, waitFor, screen, getByText} from '@testing-library/react';
import renderer from 'react-test-renderer';
import Slider from '../slider.r';
import React from 'react';

describe('Slider should', () => {

    const Slide = () => (
        <div className="slide">
            Slide Content
        </div>
    );

    test('render correctly', () => {
        const page = (
            <div>
                <Slider>
                    <Slide />
                    <Slide />
                    <Slide />
                </Slider>
            </div>
        );
        const slides = renderer.create(page);
        const snap = slides.toJSON();
        expect(snap).toMatchSnapshot();
    });
});