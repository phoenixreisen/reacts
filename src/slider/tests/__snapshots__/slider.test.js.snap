// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slider should render correctly 1`] = `
<div>
  <div
    className="swiper"
  >
    <div
      className="swiper-wrapper"
    >
      <div
        className="swiper-slide"
      >
        <div
          className="slide"
        >
          Slide Content
        </div>
      </div>
      <div
        className="swiper-slide"
      >
        <div
          className="slide"
        >
          Slide Content
        </div>
      </div>
      <div
        className="swiper-slide"
      >
        <div
          className="slide"
        >
          Slide Content
        </div>
      </div>
    </div>
    <div
      className="swiper-scrollbar swiper-scrollbar-test-slider"
    />
    <div
      className="swiper-button-prev swiper-button-prev-test-slider"
    >
      <i
        className="fas fa-arrow-circle-left"
      />
    </div>
    <div
      className="swiper-button-next swiper-button-next-test-slider"
    >
      <i
        className="fas fa-arrow-circle-right"
      />
    </div>
    <div
      className="swiper-pagination swiper-pagination-test-slider"
    />
  </div>
</div>
`;
